### PR TITLE
FEAT : Like👍 and Watchlater 🕓Functionality added

### PR DESCRIPTION
**Pull request by t007rushi**
# Features
## Like 
 - Add to Like
 - Remove from Like
## Watchlater
 - Add to Watchlater
 -  Remove from watchlater
 
 recording

### CODE REVIEW

File: src/frontend/Layout/Main/Main.jsx [hunk 1]
```diff
@@ -35,16 +35,31 @@ export const Main = () => {
             <ProtectedRoute ProtectedComp={<PlayList />}></ProtectedRoute>
           }
         />
-         <Route
+        <Route
           path="/playlist/:playlistID"
           element={
             <ProtectedRoute ProtectedComp={<PlaylistVideos />}></ProtectedRoute>
```
Comment (horsemaker) [inline src/frontend/Layout/Main/Main.jsx]: You could rather name it `RequireAuth` according to the followed convention.

File: src/frontend/components/OptionField/OptionsField.jsx [hunk 2]
```diff
@@ -5,39 +5,107 @@ import {
...
+  } = useAuth();
+  const AddToLike = async () => {
+    const data = await getService("likes", "post", tokenVal, vid);
+    likeDipatcher({ type: ADD_TO_LIKE, payload: data.likes });
+  };
+  const rmvFromLike = async () => {
+    const data = await getService("likes", "delete", tokenVal, vid);
+    likeDipatcher({ type: RMV_FROM_LIKE, payload: data.likes });
+  };
+  const AddTOWatchlater = async () => {
```
Comment (horsemaker) [inline src/frontend/components/OptionField/OptionsField.jsx]: Typo, I guess. `AddToWatchLater`

File: src/frontend/components/OptionField/OptionsField.jsx [hunk 3]
```diff
@@ -5,39 +5,107 @@ import {
...
   useOnClickOutside(optionsRef, () => setShowOptions(false));
+
+  const { watchlaterState, watchlaterDipatcher } = useWatchlater();
+  const { likeState, likeDipatcher } = useLike();
+
+  const {
+    user: { tokenVal },
+  } = useAuth();
+  const AddToLike = async () => {
+    const data = await getService("likes", "post", tokenVal, vid);
```
Comment (horsemaker) [inline src/frontend/components/OptionField/OptionsField.jsx:37]: Superb implementation of `getService`! 👌

File: src/frontend/context/like-context.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { useReducer, createContext, useContext } from "react";
+import { LikeReducer } from "../reducer/like-reducer";
+
+const LikeContext = createContext();
+
+const LikeProvider = ({ children }) => {
+  const initialState = {
```
Comment (horsemaker) [inline src/frontend/context/like-context.js]: You could write this `initialState` outside the provider & avoid unnecessary re-rendering of it.

**Review by horsemaker — APPROVED** [review]
Looks great Rushikesh! 👌
Keep it up. 👍

**Review by prathmeshjagtap — APPROVED** [review]
Code is clean 
I liked the use of getService function superb implementation
Learned  something new from this

File: src/frontend/components/OptionField/OptionsField.jsx (hunk 3 above)
Comment (t007rushi) [reply src/frontend/components/OptionField/OptionsField.jsx:37]: Thanks❤👍

File: src/frontend/context/like-context.js (hunk 4 above)
Comment (t007rushi) [reply src/frontend/context/like-context.js]: will do that

**Pull request merged**